drivers/nrf24l01p_ng/nrf24l01p_ng_netdev.c

```diff
@@ -0,0 +1,667 @@
+         * in order to receive ACKs.
+         * If node switches back to Rx mode, pipe 0 Rx address
+         * must be restored from params. */
+        nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_RX_ADDR_P0,
+                               dst_addr, dst_addr_len);
+        nrf24l01p_ng_reg8_write(dev, NRF24L01P_NG_REG_SETUP_AW,
+            NRF24L01P_NG_FLG_AW(nrf24l01p_ng_valtoe_aw(dst_addr_len)));
+    }
+    if (dev->state != NRF24L01P_NG_STATE_TX_MODE &&
+        dev->state != NRF24L01P_NG_STATE_STANDBY_2) {
+        dev->idle_state = dev->state;
+        if (dev->state != NRF24L01P_NG_STATE_STANDBY_1) {
+            nrf24l01p_ng_transition_to_standby_1(dev);
+        }
+        nrf24l01p_ng_transition_to_tx_mode(dev);
+    }
+    nrf24l01p_ng_release(dev);
+    _trigger_send(dev);
+    DEBUG("[nrf24l01p_ng] Sending %u bytes\n", pl_width);
+    return (int)pl_width;
+}
+
+/**
+ * @brief   NRF24L01+ @ref netdev_driver_t::isr
+ *
+ * @param[in] netdev        Abstract network device
+ */
+static
+void _isr(netdev_t *netdev)
+{
+    nrf24l01p_ng_t *dev = (nrf24l01p_ng_t *)netdev;
+
+    nrf24l01p_ng_acquire(dev);
+    gpio_irq_enable(dev->params.pin_irq);
+    uint8_t status = nrf24l01p_ng_get_status(dev);
+
+    if (status & NRF24L01P_NG_FLG_RX_DR) {
+        _isr_rx_dr(dev);
+    }
+    if (status & NRF24L01P_NG_FLG_MAX_RT) {
+        _isr_max_rt(dev);
+    }
+    if (status & NRF24L01P_NG_FLG_TX_DS) {
+        _isr_tx_ds(dev);
+    }
+
+    /* clear interrupt flags */
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_STATUS, &status, 1);
+
+    if (dev->state == NRF24L01P_NG_STATE_TX_MODE ||
+        dev->state == NRF24L01P_NG_STATE_STANDBY_2) {
+        /* frame in FIFO is not an ACK */
+        if (!(nrf24l01p_ng_reg8_read(dev, NRF24L01P_NG_REG_FIFO_STATUS) &
+            NRF24L01P_NG_FLG_TX_EMPTY)) {
+            nrf24l01p_ng_release(dev);
+            _trigger_send(dev);
+            return;
+        }
+    }
+    /* no more data to transmit */
+    if (dev->state != NRF24L01P_NG_STATE_STANDBY_1) {
+        nrf24l01p_ng_transition_to_standby_1(dev);
+    }
+    /* go to idle state */
+    if (dev->idle_state != NRF24L01P_NG_STATE_STANDBY_1) {
+        if (dev->idle_state == NRF24L01P_NG_STATE_POWER_DOWN) {
+            nrf24l01p_ng_transition_to_power_down(dev);
+        }
+        else {
+            dev->idle_state = NRF24L01P_NG_STATE_RX_MODE;
+            nrf24l01p_ng_transition_to_rx_mode(dev);
+        }
+    }
+    nrf24l01p_ng_release(dev);
+}
+
+/**
+ * @brief   @ref netdev_driver_t::get
+ *
+ * @param[in] netdev        Abstract network device
+ * @param[in] opt           netdev option type
+ * @param[out] val          Option value
+ * @param[in] max_len       Maximum option length
+ *
+ * @return                  Size of written option value
+ * @retval -ENOTSUP         Unsupported netdev option @p opt
+ */
+static
+int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len)
```

Comment:
`max_len` is only used in assert statements. The compiler gives a warning about the unused parameter if assert is disabled.

Reply:
Thanks. I will add a `(void)max_len;`